server.go

```diff
@@ -3509,6 +3516,12 @@ func (s *server) peerTerminationWatcher(p *peer.Brontide, ready chan struct{}) {
 
 			// The peer doesn't have an advertised address.
 			case err == errNoAdvertisedAddr:
+				// If it is an outbound peer then we fall
+				// back to the existing peer address.
+				if !p.Inbound() {
```

Comment:
Is it possible that an outbound peer gives an `errNoAdvertisedAddr`? If so, we might be iterating the loop too fast, as we `break` here, and then get the same error, then `break`, and forever...

Reply:
This breaks out of the switch statement and will result in us using the previously stored address for the peer.. So it will still hit the backoff below.

Reply:
Im not sure if this check is needed though. Put it here the original switch statement was only run if the peer was inbound and the checks below seem inbound specific

Reply:
yeah me neither, probably not, not sure...the only difference I see is whether we want to keep reconnecting when tor connection is accepted.